vault: document breaking changes to RSA key parser (#2521)
hashicorp/web-unified-docs: content/vault/global/partials/important-changes/breaking-changes/cve-2026-39829.mdx

### CVE-2026-39829: Maximum RSA key modulus size enforced ((#cve-2026-39829))

| Change       | Affected version                          | Vault edition
| ------------ | ----------------------------------------- | -------------

> | Breaking     | 1.19.18+, 1.20.12+, 1.21.7+, 2.0.2+       | All

Reviewer: ```suggestion
<!-- BEGIN: Vault:=v2.x -->

| Change       | Affected version | Vault edition
| ------------ | ---------------- | -------------
| Breaking     | 2.0.2+           | All

<!-- END: Vault:=v2.x -->

<!-- BEGIN: Vault:=v1.21.x -->

| Change       | Affected version | Vault edition
| ------------ | ---------------- | -------------
| Breaking     | 1.21.7+          | All

<!-- END: Vault:=v1.21.x -->

<!-- BEGIN: Vault:=v1.20.x -->

| Change       | Affected version | Vault edition
| ------------ | ---------------- | -------------
| Breaking     | 1.20.12+         | All

<!-- END: Vault:=v1.20.x -->

<!-- BEGIN: Vault:=v1.19.x -->

| Change       | Affected version | Vault edition
| ------------ | ---------------- | -------------
| Breaking     | 1.19.18+         | All

<!-- END: Vault:=v1.19.x -->
```

We use version gates so only the info for the relevant version shows up when you're looking at the versioned page in the docsite

> We have addressed CVE-2026-39829, a potential denial-of-service vulnerability in the RSA key parser. As part of this security fix, Vault enforces a maximum RSA key modulus size of 8192 bits, aligning with the updated golang.org/x/crypto library. 

Reviewer: ```suggestion
We have addressed CVE-2026-39829, a potential denial-of-service vulnerability in
the RSA key parser, to prioritize system stability and protection against
resource exhaustion attacks and align with the updated `golang.org/x/crypto`
library.

As part of the remediation, Vault now enforces a maximum RSA key modulus size of
8192 bits, which means you cannot use or sign RSA keys exceeding 8192 bits.
```
Style correction: write in active voice

> Users with RSA keys exceeding 8192 bits will no longer be able to use or sign these keys with Vault. RSA keys with moduli of 2048-8192 bits provide strong security while maintaining compatibility. This change prioritizes system stability and protection against resource exhaustion attacks.

Reviewer: ```suggestion
```
Folded into the paragraph above and the recommendation below

#### Recommendation

> If you use RSA keys with Vault:

Reviewer: ```suggestion
If you use RSA keys with Vault, we recommend using RSA keys with moduli of
2048-8192 bits to provide strong security while maintaining compatibility:
```
Keep recommendations in the "Recommendation" section rather than the issue description
